offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -152,42 +159,81 @@ class OfferToPDFConverter implements OfferExporter {
         // Let's clear the existing item template content first
         htmlContent.getElementById("product-items-1").empty()
         //and remove the footer on the first page
-        htmlContent.getElementById("grid-table-footer").remove()
+        //htmlContent.getElementById("grid-table-footer").remove()
         // Set the start offer position
         def itemPos = 1
+        def currTableItems = 1
         // max number of table items per page
-        def maxTableItems = 10
+        def maxTableItems = 8
         //
         def tableNum = 1
-        def elementId = "product-items"+"-"+tableNum
-        // Create the items in html in the overview table
-        offer.items.each { item ->
+        def elementId = "product-items" + "-" + tableNum
 
-            if (itemPos % maxTableItems == 0) //start (next) table
-            {
-                elementId = "product-items"+"-"+ ++tableNum
-                htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
-            }
-            htmlContent.getElementById(elementId)
-                    .append(ItemPrintout.itemInHTML(itemPos++, item))
+        List<ProductItem> listOverheadItems = offer.itemsWithOverhead
+        List<ProductItem> listNoOverheadItems = offer.itemsWithoutOverhead
 
+        // Create the items in html in the overview table
+        if (listOverheadItems.size() > 0) {
+            listOverheadItems.each { productItem ->
+                if (currTableItems >= maxTableItems) //start (next) table
+                {
+                    println(productItem.product.productName)
+                    elementId = "product-items" + "-" + ++tableNum
+                    htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
+                    currTableItems = 1
+                }
+                htmlContent.getElementById(elementId)
+                        .append(ItemPrintout.itemInHTML(itemPos++, productItem))
+                currTableItems++
+            }
+            // Add subtotal Footer to ProductItem
+            htmlContent.getElementById(elementId).append(ItemPrintout.subTotalFooter(true))
+            // Include spacing for Footer
+            currTableItems = currTableItems + 2
+            // Reset ItemPosition
+            itemPos = 1
+            //Update Pricing of Footer
+            setIntermediatePrices(true)
+        }
+        if (listNoOverheadItems.size() > 0) {
+            listNoOverheadItems.each { productItem ->
+                if (currTableItems >= maxTableItems) //start (next) table
+                {
+                    println(productItem.product.productName)
+                    elementId = "product-items" + "-" + ++tableNum
+                    htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
+                    currTableItems = 1
+                }
```

Comment:
seems like duplicate code. If you really need the exact code in this order I would suggest creating a small method that executes this. It would help the reading-flow